src/components/Tokens/TokenDetails/PriceChart.tsx

```diff
@@ -127,15 +127,26 @@ interface PriceChartProps {
 
 export function PriceChart({ width, height, tokenAddress, priceData }: PriceChartProps) {
   const [timePeriod, setTimePeriod] = useAtom(filterTimeAtom)
+  const [firstLoadedPriceDisplayed, setFirstLoadedPriceDisplayed] = useState(false)
   const locale = useActiveLocale()
   const theme = useTheme()
 
   const { priceMap } = useTokenPricesCached(priceData, tokenAddress, 'ETHEREUM', timePeriod)
   const prices = priceMap.get(timePeriod)
 
+  // first price point on the x-axis of the current time period's chart
   const startingPrice = prices?.[0] ?? DATA_EMPTY
+  // last price point on the x-axis of the current time period's chart
   const endingPrice = prices?.[prices.length - 1] ?? DATA_EMPTY
   const [displayPrice, setDisplayPrice] = useState(startingPrice)
+
+  // set display price to ending price once first set of prices is loaded.
+  useEffect(() => {
+    if (prices && !firstLoadedPriceDisplayed) {
+      setDisplayPrice(endingPrice)
+      setFirstLoadedPriceDisplayed(true)
+    }
+  }, [prices, firstLoadedPriceDisplayed, endingPrice])
```

Comment:
instead of creating firstLoadedPriceDisplayed, could we maybe just useEffect where every time prices is updated, we check if the display price isn't initialized (i.e. seeing if its equal to DATA_EMPTY), and if so, update it to be endingprice

Reply:
done, except not checking if its equal to DATA_EMPTY, want to update endingPrice when a new one is available due to changed prices.